ex-sdl-cairo-freetype-harfbuzz.c

```diff
@@ -198,9 +199,13 @@ int main () {
         /******************************************************************************/
 
         /* Blit our new image to our visible screen */ 
-        SDL_BlitSurface(sdl_surface, NULL, screen, NULL); 
-        SDL_Flip(screen); 
-
+       	SDL_SetRenderDrawColor(renderer,235, 52, 146,0);
+       	SDL_RenderClear(renderer);
+	SDL_Texture *texture = SDL_CreateTextureFromSurface(renderer,sdl_surface);
```

Comment:
This is in an loop, so should probably not be creating a whole new texture every time through I wouldn't think? Or you should be freeing it somewhere, but usually it's better to allocate once and keep re-using

Reply:
I managed to prevent memory leakage by adopting this pointers lifecycle. I'm trying to keep as much of the original code as possible. Since Cairo uses surfaces. 
new surface -> new texture -> destroy texture -> destroy surface
I doubt this is the best way to go about it but reading about texture manipulation in the documentation added more confusion.